### PR TITLE
Readtexture bug fix

### PR DESCRIPTION
**Pull request by 0871087123**
This is my fix for issue #664  
I tested texture format of GL_Rxxx such as GL_R16F, GL_R32F, GL_R8I and some else.
Also tested texture format of GL_RGxxx such as GL_RG16F, GL_RG32F, GL_RG16I, etc.
The test program is in examples/osgrtt , if texture is readed, then will no error code reported , and also can read the pixel value when model show  left bottom corner of view.
Test program is template from osgdeffered, and i left the title comment unchanged.

### CODE REVIEW

File: src/osg/Image.cpp [hunk 1]
```diff
@@ -535,9 +579,30 @@ unsigned int Image::computeNumComponents(GLenum pixelFormat)
         case(GL_ALPHA32UI_EXT): return 1;
         case(GL_ALPHA16F_ARB): return 1;
         case(GL_ALPHA32F_ARB): return 1;
+        case(GL_R16F): return 1;
         case(GL_R32F): return 1;
-        case(GL_RG): return 2;
```
Comment (mp3butcher) [inline src/osg/Image.cpp]: any reason for this removal?!

**Comment by openscenegraph** [comment]
Thanks for the changes.  Could you break this PR into two parts, one for the changes to Image.cpp and another for the new example.  This will make it clearer for others to track, and easier for me to cherry-pick the fixes into the 3.6 branch so it can be part of the upcoming 3.6.4 release.

**Comment by 0871087123** [comment]
> Thanks for the changes. Could you break this PR into two parts, one for the changes to Image.cpp and another for the new example. This will make it clearer for others to track, and easier for me to cherry-pick the fixes into the 3.6 branch so it can be part of the upcoming 3.6.4 release.

@openscenegraph  I changed this PR changing Image.cpp only, and another branch contains 
 one example testing this change still stay in my repo. Should i make a PR on example too?

**Comment by openscenegraph** [comment]
Thanks Kent.  I will go ahead and merge the Image.cpp changes to master, then have a look at if the new additions require additionally #define for older gl.h headers, then merge this with the 3.6 branch so it can be part of 3.6.4.

**Pull request merged**